test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -311,9 +312,9 @@ protected String getTestReadinessPorts() {
     public static class VersionSensitiveWarningsHandler implements WarningsHandler {
         Set<String> requiredSameVersionClusterWarnings = new HashSet<>();
         Set<String> allowedWarnings = new HashSet<>();
-        final Set<Version> testNodeVersions;
+        private final Set<String> testNodeVersions;
```

Comment:
@williamrandolph I see you have made exactly the same change; I will merge your PR in and adjust once approved.